kong/db/declarative/init.lua

```diff
@@ -802,6 +803,31 @@ function declarative.load_into_cache_with_events(entities, meta, hash)
     return nil, "failed to persist cache page number inside shdict: " .. err
   end
 
+  ngx.shared.kong:add("declarative:flips", 0, 60)
+
+  local sleep_left = 60
+  local sleep_time = 0.0375
+
+  while sleep_left > 0 do
+    local flips = ngx.shared.kong:get("declarative:flips")
+    if  flips == nil or flips == WORKER_COUNT then
```

Comment:
if i follow this correctly, the `== WORKER_COUNT` part is to avoid the case where _all_ workers are waiting for each other, right?  would it be safer to use `=>` ?

Reply:
In short, == WORKER_COUNT is the normal expacted state. nil and returning early are just unexpected timeouts.

It is used to make worker that executed this code wait for all the workers to process the flip event. Which is equal to number of workers. I am not sure if there can ve more. Crashing workers, or simething unexpected perhaps? Worker skipping a worker event? Leaking lock? We can certainly change this.